File: models/logic.py
"""Model logic - manages loaded models, registry, and multi-model switching."""
import os, json

MODELS_DIR = os.path.expanduser("~/.config/otter/models/")

def load_registry():
    registry_path = os.path.join(os.path.dirname(MODELS_DIR), "models_registry.json")
    if os.path.exists(registry_path):
        with open(registry_path) as f:
            return json.load(f)
    return {"models": {}, "count": 0}

def save_registry(registry):
    registry_path = os.path.join(os.path.dirname(MODELS_DIR), "models_registry.json")
    with open(registry_path, "w") as f:
        json.dump(registry, f, indent=2)

def register_model(file_name):
    registry = load_registry()
    registry["models"][file_name] = {"status": "loaded", "path": f"{MODELS_DIR}{file_name}"}
    registry["count"] = len(registry["models"])
    save_registry(registry)
    return f"Registered: {file_name}"

File: models/logic.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
)

type Registry struct {
	Models map[string]map[string]string `json:"models"`
	Count  int                         `json:"count"`
}

func getRegistryPath() string {
	home, _ := os.UserHomeDir()
	return filepath.Join(home, ".config", "otter", "models_registry.json")
}

func loadRegistry() Registry {
	path := getRegistryPath()
	var reg Registry
	reg.Models = make(map[string]map[string]string)

	file, err := os.ReadFile(path)
	if err == nil {
		json.Unmarshal(file, &reg)
	}
	return reg
}

func saveRegistry(reg Registry) {
	path := getRegistryPath()
	os.MkdirAll(filepath.Dir(path), 0755)
	data, _ := json.MarshalIndent(reg, "", "  ")
	os.WriteFile(path, data, 0644)
}

func registerModel(fileName string) string {
	reg := loadRegistry()
	home, _ := os.UserHomeDir()
	modelsDir := filepath.Join(home, ".config", "otter", "models")

	reg.Models[fileName] = map[string]string{
		"status": "loaded",
		"path":   filepath.Join(modelsDir, fileName),
	}
	reg.Count = len(reg.Models)
	saveRegistry(reg)
	return fmt.Sprintf("Registered: %s", fileName)
}

func main() {
	if len(os.Args) > 1 {
		fmt.Println(registerModel(os.Args[1]))
	} else {
		fmt.Println("Usage: logic <model_file_name>")
	}
}
